pkg/ingester/ingester.go

```diff
@@ -900,6 +908,10 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 		if errors.As(firstPartialErr, &ve) {
 			code = ve.code
 		}
+		var sampledErr util_log.SampledError
+		if errors.As(firstPartialErr, &sampledErr) {
+			return &mimirpb.WriteResponse{}, newErrorWithStatus(wrapWithUser(firstPartialErr, userID), code)
+		}
 		return &mimirpb.WriteResponse{}, newErrorWithStatus(annotateWithUser(firstPartialErr, userID), code)
```

Comment:
Wrapping all the errors returned by ingester could be an unsafe operation. As explained by @pracucci: "There's some context [here](https://github.com/cortexproject/cortex/pull/2004) but TL;DR is that some errors carry the series labels but these series labels are only safe to read during the execution of push() because they're unmarshalled from the write request into a pool. So before returning, we effectively "make a copy" to ensure the error will be safe after the push() has returned"

Reply:
Why is it safe to wrap the `log.SampledError` then? It also holds a reference to `err`.

Reply:
Created a related issue: https://github.com/grafana/mimir/issues/6008